skimage/morphology/watershed.py

```diff
@@ -154,17 +172,19 @@ def _offsets_to_raveled_neighbors(image_shape, structure, center, order='C'):
     return offsets[np.argsort(squared_distances)]
 
 
-def watershed(image, markers, connectivity=1, offset=None, mask=None,
+def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
               compactness=0, watershed_line=False):
     """Find watershed basins in `image` flooded from given `markers`.
 
     Parameters
     ----------
     image: ndarray (2-D, 3-D, ...) of integers
         Data array where the lowest value points are labeled first.
-    markers: int, or ndarray of int, same shape as `image`
+    markers: int, or ndarray of int, same shape as `image`, optional
         The desired number of markers, or an array marking the basins with the
-        values to be assigned in the label matrix. Zero means not a marker.
+        values to be assigned in the label matrix. Zero means not a marker. If
+        no markers are given, the local minima of the image are used as
```

Comment:
Nitpick: I think "no markers" may be confused with giving 0 or passing in an array that is False everywhere.

```suggestion
        no ``None``, the local minima of the image are used as
```

Maybe remove "Zero means not a marker." and just display a warning if the input leads to no seeds being defined. Is there a use case where this doesn't hint at a usage error?

Reply:
If ``None`` (no markers provided), the local minimal of the image...